q2_ms/utils.py

```diff
@@ -0,0 +1,61 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2025, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import subprocess
+
+EXTERNAL_CMD_WARNING = (
+    "Running external command line application(s). "
+    "This may print messages to stdout and/or stderr.\n"
+    "The command(s) being run are below. These commands "
+    "cannot be manually re-run as they will depend on "
+    "temporary files that no longer exist."
+)
+
+
+def run_command(cmd, cwd, verbose=True, env=None):
+    if verbose:
+        print(EXTERNAL_CMD_WARNING)
+        print("\nCommand:", end=" ")
+        print(" ".join(cmd), end="\n\n")
+    subprocess.run(cmd, check=True, cwd=cwd, env=env)
+
+
+def run_r_script(params, script_name, package_name):
+    """
+    Constructs a command-line call to an R script with parameters passed as
+    command-line flags and executes it.
+
+    Parameters
+        params : dict
+            A dictionary of parameter names and values to be passed as
+            command-line arguments to the R script.
+        script_name : str
+            The base name of the R script (without the `.R` extension path).
+            The script is assumed to be located at q2_ms/assets and registered in
+            pyproject.toml.
+        package_name : str
+            The name of the R package being invoked, used for error messaging.
```

Comment:
Please use the same docstring style throughout the plugin - this one here is not what we usually use.